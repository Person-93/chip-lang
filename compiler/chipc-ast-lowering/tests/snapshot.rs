use chip_snapshot_tests::{assert_snapshot, cases, run_case};
use chipc_ast::Ast;
use chipc_ast_lowering::lower_package;
use chipc_hir::HirPretty;

#[chip_test_harness::nextest_main(snapshot)]
fn main(name: &str) {
  let source = run_case(name);
  let ast = Ast::parse(&source);
  let hir = lower_package(&ast);
  let hir = format!("{}", hir.root.pretty_print(&hir).format());
  let hir = if hir.is_empty() {
    "// empty hir".to_string()
  } else {
    hir
  };
  assert_snapshot!(hir);
}
