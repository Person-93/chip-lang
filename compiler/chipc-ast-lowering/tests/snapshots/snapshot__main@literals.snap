---
source: compiler/chipc-ast-lowering/tests/snapshot.rs
assertion_line: 17
expression: hir
---
priv fn main() -> ()
  codeblock
    let number: := NumLit(42 - type:???)
    let number: := NumLit(42 - type:u8)
    let number: := NumLit(42 - type:u16)
    let number: := NumLit(42 - type:u32)
    let number: := NumLit(42 - type:u64)
    let number: := NumLit(42 - type:u128)
    let number: := NumLit(42 - type:usize)
    let number: := NumLit(42 - type:i8)
    let number: := NumLit(42 - type:i16)
    let number: := NumLit(42 - type:i32)
    let number: := NumLit(42 - type:i64)
    let number: := NumLit(42 - type:i128)
    let number: := NumLit(42 - type:isize)
    let number: := NumLit(42 - type:f32)
    let number: := NumLit(42 - type:f64)
    let number: := NumLit(42.0 - type:f??)
    let number: := NumLit(42.0 - type:f32)
    let number: := NumLit(42.0 - type:f64)
    let number: := NumLit(169 - type:???)
    let number: := NumLit(46118400291 - type:???)
    let text: := "hello world"
    let boolean: := true
    let boolean: := false
